var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -0,0 +1,288 @@
+from spack import *
+
+import socket
+import os
+
+import llnl.util.tty as tty
+
+
+def cmake_cache_entry(name, value):
+    """
+    Helper that creates CMake cache entry strings used in
+    'host-config' files.
+    """
+    return 'set("%s" "%s" CACHE PATH "")\n\n' % (name, value)
+
+
+class Conduit(Package):
+    """Conduit is an open source project from Lawrence Livermore National
+    Laboratory that provides an intuitive model for describing hierarchical
+    scientific data in C++, C, Fortran, and Python. It is used for data
+    coupling between packages in-core, serialization, and I/O tasks."""
+
+    homepage = "http://software.llnl.gov/conduit"
+    url = "https://github.com/LLNL/conduit/archive/v0.2.0.tar.gz"
+
+    version('0.2.0', 'd595573dedf55514c11d7391092fd760')
+
+    version('github-master', git='https://github.com/LLNL/conduit.git')
+
+    ###########################################################################
+    # package variants
+    ###########################################################################
+
+    variant("shared", default=True, description="Build Conduit as shared libs")
+
+    variant("cmake", default=True,
+            description="Build CMake (if off, attempt to use cmake from PATH)")
+
+    # variants for python support
+    variant("python", default=True, description="Build Conduit Python support")
+
+    # variants for i/o
+    variant("mpi", default=True, description="Build Conduit MPI Support")
+    variant("hdf5", default=True, description="Build Conduit HDF5 support")
+    variant("silo", default=True, description="Build Conduit Silo support")
+
+    # variants for dev-tools (docs, etc)
+    variant("docs", default=True,   description="Build Conduit's docs")
+
+    ###########################################################################
+    # package dependencies
+    ###########################################################################
+
+    #######################
+    # CMake
+    #######################
+    depends_on("cmake@3.3.1", when="+cmake")
+
+    #######################
+    # Python
+    #######################
+    depends_on("python", when="+python")
+    depends_on("py-numpy~blas~lapack", when="+python")
+    depends_on("py-sphinx", when="+python+docs")
+    depends_on("doxygen", when="+docs")
+
+    #######################
+    # I/O Packages
+    #######################
+    depends_on("hdf5+shared~cxx~mpi~fortran", when="+shared")
+    depends_on("hdf5~shared~cxx~mpi~fortran", when="~shared")
+
+    depends_on("silo+shared~fortran", when="+shared")
+    depends_on("silo~shared~fortran", when="~shared")
+
+    #######################
+    # MPI
+    #######################
+    depends_on("mpi", when="+mpi")
+
+    def install(self, spec, prefix):
+        """
+        Build and install Conduit.
+        """
+        with working_dir('spack-build', create=True):
+            host_cfg_fname = self.create_host_config(spec, prefix)
```

Comment:
A CMake-based package that relies on a manually-built host configuration file is highly unusual, to say the least.  Can Conduit be configured in the usual way by running `cmake`?  This is especially effective when using Spack, since Spack already prevents the "wrong" version of libraries from polluting the build by keeping the env clean and having just one package per install directory.  If you can get rid of the manually-built host config file and let CMake write that file for you, then MOST of the code in this package goes away.